module/observations/helper.js

```diff
@@ -2819,20 +2823,53 @@ module.exports = class ObservationsHelper {
           let topLevelEntityId = bodyData[topLevelEntityType];
 
 
-          let roles = bodyData.role.split(",");
+          let tenantDetails = await userService.fetchPublicTenantDetails(tenantData.tenantId);
+          if (!tenantDetails.data || !tenantDetails.data.meta || tenantDetails.success !== true) {
+            throw {
+              status: httpStatusCode.bad_request.status,
+              message: messageConstants.apiResponses.FAILED_TO_FETCH_TENANT_DETAILS,
+            };
+          }
+
+          let observableEntityKeys = tenantDetails.data.meta.observableEntityKeys;
```

Comment:
@borkarsaish65 if observableEntityKeys is not getting in meta it may throw error as undefined please rewrite like below
let observableEntityKeys = tenantDetails.data.meta?.observableEntityKeys || [];

Reply:
done